### ISSUE TITLE
Basic Setup not working: Error: The CA was unable to validate the file you provisioned

### ISSUE THREAD

**Issue by gsccheng** [Daplie/greenlock-express#109]
I'm following a generally basic setup from the docs using `greenlock-express` v2.0.9. The main difference I think is that my v4.13.4 Express on v6.10.0 Node server is not using unprivileged ports (listening on 8080 for port forwarded https connections and 8081 for port forwarded http connections). I'm not using any other web server in front, and the user connects through a Kubernetes service that maps the ports (e.g. 443:8080, 80:8081) to the docker container/pod that holds this Node server.

There's not any other configurations (e.g. directory setup, privilege changes, installing other modules/plugins) I'm supposed to do right since that's supposed to be handled out of the box? Also, the email in opts.email should match the email in [https://whois.icann.org](https://whois.icann.org) right?

Upon starting the server, I am hitting the listener callbacks, in addition to this warning that I've read is normal since I'm not listening on ports 443 or 80:
```
le.challenges[tls-sni-01].loopback should be defined as function (opts, domain, token, cb) { ... } and should prove (by external means) that the ACME server challenge 'tls-sni-01' will succeed
```

**server.js**
```
import Lex from 'greenlock-express';
import https from 'https';
import http from 'http';

function approveDomains(opts, certs, cb) {
  console.log('opts are', opts);
  if (certs) {
    opts.domains = certs.altnames;
  }
  else {
    opts.domains = ['mydomainname.com', 'www.mydomainname.com'];
    opts.email = 'mydomainname.com@domainsbyproxy.com';
    opts.agreeTos = true;
  }

  cb(null, { options: opts, certs: certs });
}

const lex = Lex.create({
  server: 'https://acme-v01.api.letsencrypt.org/directory',
  approveDomains
})

const app = new Express();

http.createServer(lex.middleware(require('redirect-https')())).listen(8081, function () {
    console.log("Listening for ACME http-01 challenges on", "http://mydomainname.com", 8081);
  });

  https.createServer(lex.httpsOptions, lex.middleware(app)).listen(config.port, function (err) {
    console.log("Listening for ACME tls-sni-01 challenges and serve app on", "http://mydomainname.com", 8080);
  });
```

**Error logs from server:**
```
0] ERROR in le-sni-auto:
[0] Error: The CA was unable to validate the file you provisioned.
[0]  - https://acme-v01.api.letsencrypt.org/acme/challenge/AeZZYbJ3TZmWGhUNzQo85mxBQHl7F-2Fz1kMP3RSHpo/837151347 [invalid]
[0]    Could not connect to mydomainname.com
[0]  - https://acme-v01.api.letsencrypt.org/acme/challenge/AeZZYbJ3TZmWGhUNzQo85mxBQHl7F-2Fz1kMP3RSHpo/837151348 [pending]
[0]  - https://acme-v01.api.letsencrypt.org/acme/challenge/AeZZYbJ3TZmWGhUNzQo85mxBQHl7F-2Fz1kMP3RSHpo/837151349 [pending]
[0]     at handleErr (/usr/src/app/node_modules/le-acme-core/lib/get-certificate.js:351:17)
[0]     at ensureValidation (/usr/src/app/node_modules/le-acme-core/lib/get-certificate.js:226:16)
[0]     at Request._callback (/usr/src/app/node_modules/le-acme-core/lib/get-certificate.js:208:13)
[0]     at Request.self.callback (/usr/src/app/node_modules/request/request.js:188:22)
[0]     at emitTwo (events.js:106:13)
[0]     at Request.emit (events.js:191:7)
[0]     at Request.<anonymous> (/usr/src/app/node_modules/request/request.js:1171:10)
[0]     at emitOne (events.js:96:13)
[0]     at Request.emit (events.js:188:7)
[0]     at IncomingMessage.<anonymous> (/usr/src/app/node_modules/request/request.js:1091:12)
[0]     at IncomingMessage.g (events.js:291:16)
[0]     at emitNone (events.js:91:20)
[0]     at IncomingMessage.emit (events.js:185:7)
[0]     at endReadableNT (_stream_readable.js:974:12)
[0]     at _combinedTickCallback (internal/process/next_tick.js:74:11)
[0]     at process._tickCallback (internal/process/next_tick.js:98:9)
```

If I were to just start my Node server on localhost, and `curl https://localhost:8080`, I would get basically the same errors (makes sense that it wouldn't work, but I want to point out that the errors are basically the same). However, `curl http://localhost:8081`, I would get the expected
```
<html>
<head>
  <META http-equiv="refresh" content="0;URL='https://localhost:443/'">
</head>
<body>
<!-- Hello Mr Developer! We don't serve insecure resources around here.
    Please use HTTPS instead. -->
</body>
</html>
```

Unfortunately I can't test http://mydomainname.com where I'm not getting any response from my server at all, but that may be another problem:
```
curl http://mydomainname.com
```
gives
```
curl: (52) Empty reply from server
```
Thanks!

**Comment by insightfuls** [comment]
The "that may be another problem" is what you need to sort out.

What's going on with greenlock-express is that it's passing on an error message from LetsEncrypt itself. LetsEncrypt ("The CA") has attempted to connect to your server to validate the challenge:

```
[0] Error: The CA was unable to validate the file you provisioned.
...
[0]    Could not connect to mydomainname.com
```

I think it's trying to access http://mydomainname.com/.well-known/acme-challenge/AeZZYbJ3TZmWGhUNzQo85mxBQHl7F-2Fz1kMP3RSHpo

So get things working so you can curl that, and you'll have a chance.

You probably want to use the staging LetsEncrypt server, not the live one, or you'll hit your quota while you try to debug this.

**Comment by gsccheng** [comment]
Thanks for your quick response...

Yeah you're right! Turns out it was a problem with my service, where I was actually mapping http to 8080 instead of 8081 ![image](https://cloud.githubusercontent.com/assets/616415/24116780/138a2a64-0d65-11e7-8d5e-e59e902cbe3c.png)
Unfortunately I didn't catch it earlier because the basic diagnostic tools only showed where it mapped from, and I was thrown off by how I saw 443 connects (so it should still work for that case), and in testing locally it gave the same errors.

So, I only had to literally change one character and now I see my "brand new shiny certs". Thanks again!
